_sources/Functions/TypeAnnotations.rst

```diff
@@ -0,0 +1,78 @@
+.. qnum::
+   :prefix: func-annotate
+   :start: 1
+
+.. index:: 
+   type annotation
+   
+Type Annotations
+----------------
+
+In the previous section, we discussed the decoding work that is required when you look at a function and 
+are trying to determine the types of its parameters. In this section, we'll introduce a feature that can help
+reduce the amount of sleuthing that is needed.
+
+Consider the following function definition:
+
+.. sourcecode:: python
+
+    def duplicate(msg):
+        """Returns a string containing two copies of `msg`"""
+
+        return msg + msg
+
+This function is intended to duplicate a message; if called with the value 'Hello', it returns the value
+'HelloHello'. If called with other types of data, however, it will not work properly. (What will the
+function do if given an ``int`` or a ``float`` value?)
+
+Python allows you to indicate the intended type of the function parameters and the type of the function return value
+in a function definition using a special notation demonstrated in this example:
+
+.. sourcecode:: python
```

Comment:
You could make this a ..activecode:: directive as skulpt will cheerfully parse but ignore the annotations.  Which helps reinforce your message.